app/controllers/contributions_controller.rb

```diff
@@ -1,5 +1,30 @@
 class ContributionsController < ApplicationController
+  before_action :authenticate_user!
+
   def new
-    
+    @project = project
+    @contribution = Contribution.new(user: current_user, project: @project)
   end
+  
+  def create
+    @project = project
+    result = Contributions::CreateTransaction.call(contribution_params.merge(project: @project, user: current_user))
+    if result.success
+      redirect_to users_dashboards_path
+    else
+      @contribution = result.failure[:contribution]
+      render :new
+    end
+  end
+
+  private
+  
+  def contribution_params
+    params.require(:contribution).permit(:amount, :counterpart_id)
+  end
+
+  def project
+    @decorated_project ||= Project.find(params[:project_id]).decorate
+  end
```

Comment:
https://github.com/drapergem/draper#when-to-decorate-objects

Re-relis ce paragraphe. Re-relis ton code. Tu ne dois pas utiliser l'objet décoré dans ton controller.

Reply:
Ok